hw_diag/utilities/event_streamer.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime
+from enum import Enum, auto
+import json
+import os
+import requests
+import logging
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+VOLUME_PATH = '/var/watchdog/'
+
+NETWORK_EVENT_BASE = 0
+CONTAINER_EVENT_BASE = 1000
+
+NETWORK_ACTION_BASE = 10000
+CONTAINER_ACTION_BASE = 11000
+
+BUCKET_NAME = os.getenv(
+    'MINER_EVENTS_GCS_BUCKET',
+    'helium-miner-events'
+)
+
+URL = 'https://www.googleapis.com/upload/storage/v1/b/%s/o?uploadType=media' \
+    % BUCKET_NAME
+
+
+event_queue = {}
+EVENT_TYPE_KEY = 'event_type'
+ACTION_TYPE_KEY = 'action_type'
+
+
+class DiagEvent(Enum):
+    NETWORK_DISCONNECTED = NETWORK_ACTION_BASE  # completely disconnected.
+    NETWORK_LOCAL_CONNECTED = auto()  # connected to local gateway
```

Comment:
was a little torn between whether all events and actions should be defined here or every watchdog should define their own events. In the end kept it at a single place.